post-scan-actions/gcp-python-promote-or-quarantine/main.tf

```diff
@@ -0,0 +1,88 @@
+terraform {
+  required_providers {
+    google = {
+      source = "hashicorp/google"
+      version = "~> 4.0.0"
+    }
+  }
+}
+
+provider "google" {
+  project = var.project_settings.project_id
+  region = var.project_settings.region
+}
+
+resource "random_id" "deploy_suffix" {
+  keepers = {
+    # Generate a new suffix each time we switch to a new deployment
+    deployment_name = "${var.deployment_name}"
+  }
+
+  byte_length = 5
+}
+
+resource "google_service_account" "service_account" {
+  account_id = "qua-pro-plugin-sa-${lower(random_id.deploy_suffix.hex)}"
+  display_name = "Quarantine/Promote plugin Service Account"
+  project = var.project_settings.project_id
+}
+
+resource "google_project_iam_custom_role" "storage_bucket_access" {
+  role_id = "plugin_storage_bucket_access_${lower(random_id.deploy_suffix.hex)}"
+  title = "Storage Bucket Access"
+  permissions = var.promote_mode == "move" || var.quarantine_mode == "move" ? [
+    "storage.objects.create",
+    "storage.objects.delete",
+    "storage.objects.get",
+    "storage.objects.update"
+  ] : [
+    "storage.objects.create",
+    "storage.objects.get",
+    "storage.objects.update"
+  ]
+}
+
+resource "google_project_iam_member" "storage_bucket_role_binding" {
+  project = var.project_settings.project_id
+  role    = google_project_iam_custom_role.storage_bucket_access.name
+  member  = "serviceAccount:${google_service_account.service_account.email}"
+}
```

Comment:
Would [google_storage_bucket_iam_binding](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/storage_bucket_iam#google_storage_bucket_iam_binding) be more suitable for limiting the permissions to certain bucket?

Reply:
Yes I think it would be more suitable.